sqlmesh/cli/main.py

```diff
@@ -303,6 +303,12 @@ def diff(ctx: click.Context, environment: t.Optional[str] = None) -> None:
     help="Create a plan for forward-only changes.",
     default=None,
 )
+@click.option(
+    "--allow-destructive-model",
```

Comment:
Do we need a selector here? Shouldn't this be a global plan flag `--allow-descrtuctive`?

Reply:
I think selector seems correct since it requires the user to be intentional about what they are allowing. Even if it does require some extra work on their end I think of that as a feature rather than something to make easier.